### PR TITLE
docs(suspensive.org): add docs for shouldCatch at `ErrorBoundary`

### LINKED ISSUE #952: [Feature]: add doc for shouldCatch prop
### Package Scope

@suspensive/react

### Description

add doc for shouldCatch prop

### Possible Solution

_No response_

### etc.

_No response_

### PR DESCRIPTION
**Pull request by jgjgill**
# Overview

close #952 

Hi, I worked on the docs. 🧐
This is my first time adding docs to this library.
So please feel free to tell me if I'm missing anything.

Thanks for the opportunity. 🙇‍♂️

## PR Checklist

- [x] I did below actions if need

1. I read the [Contributing Guide](https://github.com/toss/suspensive/blob/main/CONTRIBUTING.md)
2. I added documents and tests.

### CODE REVIEW

File: docs/suspensive.org/src/pages/docs/react/ErrorBoundary.en.mdx [hunk 1]
```diff
@@ -364,6 +365,226 @@ export const ErrorAfter2s = () => {
...
+### props.shouldCatch
+
+shouldCatch determines whether `<ErrorBoundary/>` should catch errors based on conditions.
+
+It accepts three criteria: Boolean, Constructor, and Callback, and defaults to true.
+
+<Tabs items={['Boolean', 'Constructor', 'Callback']}>
+  <Tabs.Tab>
+  
+```tsx {10}
```
Comment (manudeli) [inline docs/suspensive.org/src/pages/docs/react/ErrorBoundary.en.mdx:377]: I think {10} is good, but /shouldCatch/ also could be same benefit and good to maintain 

```suggestion
```tsx /shouldCatch/
```

File: docs/suspensive.org/src/pages/docs/react/ErrorBoundary.en.mdx [hunk 2]
```diff
@@ -364,6 +365,226 @@ export const ErrorAfter2s = () => {
...
+        <CustomErrorAfter2s />
+      </ErrorBoundary>
+    </ErrorBoundary>
+  )
+}
+```
+  </Tabs.Tab>
+  <Tabs.Tab>
+
+```tsx {10}
```
Comment (manudeli) [inline docs/suspensive.org/src/pages/docs/react/ErrorBoundary.en.mdx:399]:
```suggestion
```tsx /shouldCatch/
```

File: docs/suspensive.org/src/pages/docs/react/ErrorBoundary.en.mdx [hunk 3]
```diff
@@ -364,6 +365,226 @@ export const ErrorAfter2s = () => {
...
+  >({
+    isError: false,
+    error: null
+  });
+
+  useEffect(() => {
+    setTimeout(() => {
+      setAsyncState({
+        isError: true,
+        error: () => new CustomError("unauthorized")
```
Comment (manudeli) [inline docs/suspensive.org/src/pages/docs/react/ErrorBoundary.en.mdx:514]: Could you remove all unnecessary context in this doc please? In my opinion, text "unauthorized" could make misunderstanding concept shouldCatch

```suggestion
        error: () => new CustomError("⚠️ Error made by CustomError Constructor")
```

File: docs/suspensive.org/src/pages/docs/react/ErrorBoundary.en.mdx [hunk 4]
```diff
@@ -364,6 +365,226 @@ export const ErrorAfter2s = () => {
...
+
+export const ErrorAfter2s = () => {
+  const [asyncState, setAsyncState] = useState<{ isError: true; error: Error } | { isError: false; error: null }>({
+    isError: false,
+    error: null,
+  })
+
+  useEffect(() => {
+    setTimeout(() => {
+      setAsyncState({ isError: true, error: { status: 401, message: 'unauthorized' } })
```
Comment (manudeli) [inline docs/suspensive.org/src/pages/docs/react/ErrorBoundary.en.mdx:574]: Same here

File: docs/suspensive.org/src/pages/docs/react/ErrorBoundary.en.mdx [hunk 5]
```diff
@@ -364,6 +365,226 @@ export const ErrorAfter2s = () => {
...
 </Sandpack>
 
+### props.shouldCatch
+
+shouldCatch determines whether `<ErrorBoundary/>` should catch errors based on conditions.
+
+It accepts three criteria: Boolean, Constructor, and Callback, and defaults to true.
+
+<Tabs items={['Boolean', 'Constructor', 'Callback']}>
+  <Tabs.Tab>
```
Comment (manudeli) [inline docs/suspensive.org/src/pages/docs/react/ErrorBoundary.en.mdx:375]:
```suggestion
It accepts three criteria: Boolean, ErrorConstructor, and Callback, and defaults to true.

<Tabs items={['Boolean', 'ErrorConstructor', 'Callback']}>
  <Tabs.Tab>
```

File: docs/suspensive.org/src/pages/docs/react/ErrorBoundary.en.mdx [hunk 6]
```diff
@@ -364,6 +365,226 @@ export const ErrorAfter2s = () => {
...
+  }, []);
+
+  if (asyncState.isError) {
+    throw asyncState.error();
+  }
+
+  return <>No error</>;
+};
+```
+
```
Comment (manudeli) [inline docs/suspensive.org/src/pages/docs/react/ErrorBoundary.en.mdx:526]: In my opinion, no depth and one file could make doc easier to understand .

```suggestion
```tsx Example.tsx active
import { useState, useEffect } from "react";
import { ErrorBoundary } from '@suspensive/react'
import { createElement } from 'react'

export const Example = () => {
  return (
    <ErrorBoundary fallback={({ error }) => <>{error.message} of Parent</>}>
      <ErrorBoundary
        shouldCatch={CustomError}
        fallback={({ error }) => <>{error.message} of Child</>}
      >
        <CustomErrorAfter2s />
      </ErrorBoundary>
    </ErrorBoundary>
  )
}

class CustomError extends Error {
  constructor(...args: ConstructorParameters<ErrorConstructor>) {
    super(...args)
    console.error(...args)
  }
}

export const CustomErrorAfter2s = () => {
  const [asyncState, setAsyncState] = useState<
    { isError: true; error: CustomError } | { isError: false; error: null }
  >({
    isError: false,
    error: null
  });

  useEffect(() => {
    setTimeout(() => {
      setAsyncState({
        isError: true,
        error: () => new CustomError("unauthorized")
      });
    }, 2000);
  }, []);

  if (asyncState.isError) {
    throw asyncState.error();
  }

  return <>No error</>;
};
```

```

**Review by manudeli — COMMENTED** [review]
@jgjgill Thanks!! You made cool example too!
I left comments to improve some things

File: docs/suspensive.org/src/pages/docs/react/ErrorBoundary.en.mdx (hunk 1 above)
Comment (jgjgill) [reply docs/suspensive.org/src/pages/docs/react/ErrorBoundary.en.mdx]: oh, I think it's definitely better. 👍

File: docs/suspensive.org/src/pages/docs/react/ErrorBoundary.en.mdx (hunk 3 above)
Comment (jgjgill) [reply docs/suspensive.org/src/pages/docs/react/ErrorBoundary.en.mdx]: I agree, I changed it!

File: docs/suspensive.org/src/pages/docs/react/ErrorBoundary.en.mdx (hunk 6 above)
Comment (jgjgill) [reply docs/suspensive.org/src/pages/docs/react/ErrorBoundary.en.mdx]: Oh, Okay. I fixed it. 🧐

**Comment by jgjgill** [comment]
@manudeli 
Thank you for your quick feedback.
I've reflected review. 🙇‍♂️

**Review by manudeli — APPROVED** [review]
Thank you for contributing to making this library easier! I'm glad that library users can now better understand and use the shouldCatch prop. 🚀

**Pull request merged**